Makefile

```diff
@@ -1,56 +1,72 @@
-default: prepare-build
-default: build/holo build/holo-files
-default: build/man/holorc.5 build/man/holo-plugin-interface.7 build/man/holo-test.7 build/man/holo.8 build/man/holo-files.8
+bins = holo holo-files
+mans = holorc.5 holo-plugin-interface.7 holo-test.7 holo.8 holo-files.8
 
-VERSION := $(shell ./util/find_version.sh)
+default: $(addprefix bin/,$(bins)) $(addprefix man/,$(mans))
+.PHONY: default
 
-GO_BUILDFLAGS :=
-GO_LDFLAGS    := -s -w
+GO_BUILDFLAGS =
+GO_LDFLAGS    = -s -w
+GO_TESTFLAGS  = -covermode=count
 
-prepare-build:
-	@mkdir -p build/man
-build/holo: src/holo/main.go src/holo/*/*.go
-	go build $(GO_BUILDFLAGS) --ldflags "$(GO_LDFLAGS) -X main.version=$(VERSION)" -o $@ $<
-build/holo-files: src/holo-files/main.go src/holo-files/*/*.go
-	go build $(GO_BUILDFLAGS) --ldflags "$(GO_LDFLAGS)" -o $@ $<
+src/holo/version.go: FORCE
+	printf 'package main\n\nconst version = "%s"\n' "$$( ./util/find_version.sh)" | util/write-ifchanged $@
+
+$(addprefix %/,$(bins)): FORCE src/holo/version.go
+	GOPATH=$(abspath .) go install $(GO_BUILDFLAGS) --ldflags '$(GO_LDFLAGS)' $(bins)
+bin/%.test: bin/% src/main.go.test
+	GOPATH=$(abspath .) go test -c -o $@ $(GO_TESTFLAGS) -coverpkg $*/... $*
+
+man:
+	@mkdir $@
 
 # manpages are generated using pod2man (which comes with Perl and therefore
 # should be readily available on almost every Unix system)
-build/man/%: doc/%.pod
+man/%: doc/%.pod | man
 	pod2man --name="$(shell echo $* | cut -d. -f1)" --section=$(shell echo $* | cut -d. -f2) \
 		--center="Configuration Management" --release="Holo $(VERSION)" \
```

Comment:
The `$(VERSION)` interpolation is broken after you changed the usage of `util/find_version.sh`.